oasis/models/rules.py

```diff
@@ -2,15 +2,167 @@
 
 from __future__ import absolute_import
 
-from oasis.models.model import Model
+import datetime
+import time
+from threading import Thread, Timer
+from pytimeparse.timeparse import timeparse
+from oasis.models.model import Model, Config
 from oasis.datasource import PrometheusAPI, PrometheusQuery, Metrics
+from oasis.libs.log import logger
+from oasis.models.util import sub_id
+from oasis.libs.features import Features
+
+OPERATORS = ["==", "!=", "<=", ">=", "<", ">"]
 
 
 class Rules(Model):
```

Comment:
I am a little confused by Rules. Is it just rules  or it is object contains rules. From the code it seems it runs job and map with rules.

Reply:
yep, this rules just define for metric, if the metric doesn't match the rule, we can judge the metric is not abnormal. this model just do it